component/src/governance/check.rs

```diff
@@ -205,6 +251,39 @@ pub mod stateful {
         Ok(())
     }
 
+    pub async fn delegator_vote<S: StateRead>(
+        state: S,
+        DelegatorVote {
+            body:
+                DelegatorVoteBody {
+                    proposal,
+                    vote: _, // All votes are valid, so we don't need to do anything with this
+                    start_position,
+                    value,
+                    unbonded_amount,
+                    nullifier,
+                    rk: _, // We already used this to check the auth sig in stateless verification
+                },
+            auth_sig: _, // We already checked this in stateless verification
+            proof: _,    // We already checked this in stateless verification
+        }: &DelegatorVote,
+    ) -> Result<()> {
+        proposal_voteable(&state, *proposal).await?;
+        // TODO: once epochs are non-constant, this has to be re-done to be a stateful lookup:
```

Comment:
I don't think this refactor needs to be done before merging, but I think it could usefully be done on its own prior to switching to non-constant epochs, by (1) recording a `start_position` in the state for each proposal and (2) querying it here.

Then even before epochs are made non-constant, this part of the code won't depend on epoch duration math, it just has to do a lookup.